tests/test_butler.py

```diff
@@ -589,6 +589,21 @@ def testPruneCollections(self):
         ref1 = butler.put(metric, datasetType, {"instrument": "Cam1", "physical_filter": "Cam1-G"}, run=run1)
         ref2 = butler.put(metric, datasetType, {"instrument": "Cam1", "physical_filter": "Cam1-G"}, run=run2)
         ref3 = butler.put(metric, datasetType, {"instrument": "Cam1", "physical_filter": "Cam1-R1"}, run=run1)
+
+        # Add a new dataset type and delete it
+        tmpName = "prune_collections_disposable"
+        tmpDatasetType = self.addDatasetType(tmpName, dimensions, storageClass,
+                                             butler.registry)
+        tmpFromRegistry = butler.registry.getDatasetType(tmpName)
+        self.assertEqual(tmpDatasetType, tmpFromRegistry)
+        butler.registry.pruneDatasetType(tmpName)
+        with self.assertRaises(KeyError):
+            butler.registry.getDatasetType(tmpName)
+
+        # Try and fail to delete a datasetType that is associated with data
+        with self.assertRaises(Exception):
+            butler.registry.pruneDatasetType(datasetType.name)
```

Comment:
If you can think of an easy way to do it, it'd also be nice to check here that nothing did get deleted.  I'm not at all worried about the current implementation, but a different caching model might lead to an implementation where that's less obvious.